### PR TITLE
Feature/pim 1641 1643

### PR DESCRIPTION
**Pull request by vivid-cpreston**

### CODE REVIEW

File: .github/workflows/openshift-deploy.yml [hunk 1]
```diff
@@ -133,6 +150,10 @@ jobs:
           #Jasper variables
           JASPER_URL: https://cirras-underwriting-jasper-${{ inputs.ENVIRONMENT_NAME }}-${{ inputs.NAMESPACE }}.apps.silver.devops.gov.bc.ca/jasperserver/
           JASPER_USERNAME: ${{ vars.JASPER_USERNAME }}
+          JASPER_CPU_REQUEST: ${{ vars.JASPER_CPU_REQUEST }}
```
Comment (dpitk2222) [inline .github/workflows/openshift-deploy.yml:153]: I thought we don't use this action for Jasper?

File: openshift/cirras-underwriting-api-configmap.yaml [hunk 2]
```diff
@@ -5,6 +5,9 @@ metadata:
   namespace: '#{NAMESPACE}#'
 immutable: false
 data:
+  TOMCAT_PORT: '#{TARGET_PORT}#'
```
Comment (dpitk2222) [inline openshift/cirras-underwriting-api-configmap.yaml:8]: Definitely reads from TARGET_PORT, not TOMCAT_PORT?

File: openshift/cirras-underwriting-api-configmap.yaml [hunk 3]
```diff
@@ -5,6 +5,9 @@ metadata:
...
+  TOMCAT_MAX_THREADS: '#{TOMCAT_MAX_THREADS}#'
```
Comment (dpitk2222) [inline openshift/cirras-underwriting-api-configmap.yaml:9]: I don't see a TOMCAT_MAX_THREADS or CONNECTION_TIMEOUT in openshift-deploy.yml? Or is that not where they come from?

File: openshift/cirras-underwriting-ui-configmap.yaml [hunk 4]
```diff
@@ -14,6 +14,9 @@ metadata:
       fieldsType: FieldsV1
 immutable: false
 data:
+  TOMCAT_PORT: '#{TARGET_PORT}#'
```
Comment (dpitk2222) [inline openshift/cirras-underwriting-ui-configmap.yaml:17]: Same questions as for API.

File: openshift/cirras-underwriting-ui-deployment.yaml [hunk 5]
```diff
@@ -17,12 +17,27 @@ spec:
...
+            '#{IMAGE_BY_DIGEST}#'
           imagePullPolicy: Always
           ports:
             - containerPort: #{TARGET_PORT}#
               protocol: TCP
           env:
+            - name: TOMCAT_PORT
+              valueFrom: 
+                configMapKeyRef:
+                  name: cirras-underwriting-api-config-#{ENV}#
```
Comment (dpitk2222) [inline openshift/cirras-underwriting-ui-deployment.yaml:29]: Wrong config map. Should be UI, not API.

File: cirras-underwriting-war/deploy-tools/server.xml [hunk 6]
```diff
@@ -12,14 +12,14 @@
   </GlobalNamingResources>
 
   <Service name="Catalina">
-    <Connector port="#{TOMCAT_PORT}#"
+    <Connector port="${TOMCAT_PORT}"
       protocol="org.apache.coyote.http11.Http11NioProtocol"
-      maxThreads="#{MAX_THREADS}#"
-        connectionTimeout="#{CONNECTION_TIMEOUT}#"
+      maxThreads="${MAX_THREADS}"
```
Comment (dpitk2222) [inline cirras-underwriting-war/deploy-tools/server.xml:17]: This is called TOMCAT_MAX_THREADS in the config map. Or is that not where it comes from?

File: cirras-underwriting-api/deploy-tools/server.xml [hunk 7]
```diff
@@ -11,14 +11,14 @@
   </GlobalNamingResources>
 
   <Service name="Catalina">
-    <Connector port="#{TOMCAT_PORT}#"
+    <Connector port="${TOMCAT_PORT}"
       protocol="org.apache.coyote.http11.Http11NioProtocol"
-      maxThreads="#{MAX_THREADS}#"
-        connectionTimeout="#{CONNECTION_TIMEOUT}#"
+      maxThreads="${MAX_THREADS}"
```
Comment (dpitk2222) [inline cirras-underwriting-api/deploy-tools/server.xml:16]: This is called TOMCAT_MAX_THREADS in the config map. Or is that not where it comes from?

File: .github/workflows/build-one.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        required: true
+        type: string
+        description: Additional tag to apply to images being built
+        default: latest
+      COMPONENT:
+        required: true
+        type: choice
+        options:
+          - cirras-underwriting-api
+          - cirras-underwriting-war
```
[the remark this answers is not in the document]
Comment (vivid-cpreston) [reply .github/workflows/build-one.yml:16]: Doing a little refactor to get this in

File: .github/workflows/openshift-deploy.yml (hunk 1 above)
Comment (vivid-cpreston) [reply .github/workflows/openshift-deploy.yml:153]: Liquibase is the one we don't use this for - Jasper is correct here

File: .github/workflows/build-one.yml [hunk 9]
```diff
@@ -24,29 +25,23 @@ jobs:
...
   build-and-upload-npm-modules:
     uses: ./.github/workflows/build-npm.yml
-    if: ${{ inputs.COMPONENT == 'cirras-underwriting-api' }}
+    if: ${{ inputs.COMPONENT == 'cirras-underwriting-ngclient-lib' }}
     secrets: inherit
-    needs: build-underwriting-api
     with:
       ENVIRONMENT: DEV
       MICROSERVICE_NAME: cirras-underwriting-ngclient-lib
-      DOWNLOAD_FILE: CIRRAS_UNDERWRITING_API_swagger.json
```
Comment (dpitk2222) [inline .github/workflows/build-one.yml]: Missing the new API_TARGET and SWAGGER_TARGET_PATH inputs?

File: .github/workflows/build-one.yml [hunk 10]
```diff
@@ -24,29 +25,23 @@ jobs:
...
-      DOWNLOAD_PATH: cirras-underwriting-ngclient-lib/cirras-underwriting-ngclient-lib/src/main/angular/src/swagger/
       EXPORT_MODULE: cirras-underwriting-api
```
Comment (dpitk2222) [inline .github/workflows/build-one.yml:38]: Should EXPORT_MODULE be removed, like it was in build-dev?

File: openshift/cirras-underwriting-api-configmap.yaml (hunk 3 above)
Comment (dpitk2222) [reply openshift/cirras-underwriting-api-configmap.yaml:9]: Looks like TOMCAT_MAX_THREADS is fixed, but what about CONNECTION_TIMEOUT?

File: openshift/cirras-underwriting-ui-configmap.yaml [hunk 11]
```diff
@@ -14,6 +14,9 @@ metadata:
       fieldsType: FieldsV1
 immutable: false
 data:
+  TOMCAT_PORT: '#{TOMCAT_PORT}#'
+  TOMCAT_MAX_THREADS: '#{TOMCAT_MAX_THREADS}#'
+  CONNECTION_TIMEOUT: '#{CONNECTION_TIMEOUT}#'
```
Comment (dpitk2222) [inline openshift/cirras-underwriting-ui-configmap.yaml:19]: There is no CONNECTION_TIMEOUT in openshift-deploy?

File: openshift/cirras-underwriting-api-configmap.yaml (hunk 3 above)
Comment (vivid-cpreston) [reply openshift/cirras-underwriting-api-configmap.yaml:9]: Added missing property to openshift-deploy.yml

File: openshift/cirras-underwriting-ui-configmap.yaml (hunk 11 above)
Comment (vivid-cpreston) [reply openshift/cirras-underwriting-ui-configmap.yaml:19]: Added missing property to openshift-deploy.yml

**Review by dpitk2222 — APPROVED** [review]

**Pull request merged**